hydradx/model/amm/money_market.py

```diff
@@ -0,0 +1,492 @@
+        )
+
+    def copy(self):
+        return CDP(
+            self.debt,
+            self.collateral,
+            self.liquidation_threshold
+        )
+
+    def validate(self) -> bool:
+        if min(self.debt.values()) < 0 or min(self.collateral.values()) < 0:
+            return False
+        # if sorted(list(self.debt_assets.keys())) == sorted(list(self.collateral_assets.keys)):
+        #     return False
+        return True
+
+
+class MoneyMarketAsset:
+    def __init__(
+            self,
+            name: str,
+            price: float,
+            liquidity: float,
+            liquidation_bonus: float,
+            liquidation_threshold: float,
+            ltv: float,
+            emode_liquidation_bonus: float = 0,
+            emode_liquidation_threshold: float = 0,
+            emode_ltv: float = 0,
+            emode_label: str = '',
+    ):
+        self.name = name
+        self.price = price
+        self.liquidity = liquidity
+        self.liquidation_bonus = liquidation_bonus
+        self.liquidation_threshold = liquidation_threshold
+        self.ltv = ltv
+        self.emode_liquidation_bonus = emode_liquidation_bonus
+        self.emode_liquidation_threshold = emode_liquidation_threshold
+        self.emode_ltv = emode_ltv
+        self.emode_label = emode_label
+
+
+class MoneyMarket:
+    def __init__(
+            self,
+            assets: list[MoneyMarketAsset],
+            cdps: list = None,
+            full_liquidation_threshold: float = 0.95,
+            close_factor: float = 0.5
+    ):
+        self.liquidity = {
+            asset.name: asset.liquidity
+            for asset in assets
+        }
+        self.borrowed = {asset: 0 for asset in self.liquidity}
+        self.asset_list = list(self.liquidity.keys())
+        self.prices = {
+            asset.name: asset.price
+            for asset in assets
+        }
+        self.assets = assets
+        self.ltv = {
+            (asset1.name, asset2.name):
+            asset1.emode_ltv if asset1.emode_label and asset1.emode_label == asset2.emode_label else asset1.ltv
+            for asset1 in assets
+            for asset2 in assets if asset2 != asset1
+        }
+        self.liquidation_bonus = {
+            (asset1.name, asset2.name):
+            asset1.emode_liquidation_bonus if asset1.emode_label and asset1.emode_label == asset2.emode_label else asset1.liquidation_bonus
+            for asset1 in assets
+            for asset2 in assets if asset2 != asset1
+        }
+        self.liquidation_threshold = {
+            (asset1.name, asset2.name):
+            asset1.emode_liquidation_threshold if asset1.emode_label and asset1.emode_label == asset2.emode_label else asset1.liquidation_threshold
+            for asset1 in assets
+            for asset2 in assets if asset2 != asset1
+        }
+
+        self.cdps: list[CDP] = [] if cdps is None else cdps
+        for cdp in self.cdps:
+            if not cdp.liquidation_threshold:
```

Comment:
Does it make sense for a cdp to have a liquidation_threshold already that we do not overwrite here? Maybe we want to always overwrite? Same question for health_factor